dbt/adapters/athena/impl.py

```diff
@@ -833,3 +835,26 @@ def is_list(self, value: Any) -> bool:
         a list since this is complicated with purely Jinja syntax.
         """
         return isinstance(value, list)
+
+    @available
+    def safe_run_query(self, sql: str) -> str:
```

Comment:
Maybe a more understandable name and less generic like `run_query_with_partitions_limit_catching` ?